pylint/reporters/text.py

```diff
@@ -187,9 +188,21 @@ def __init__(self, output: Optional[TextIO] = None) -> None:
     def on_set_current_module(self, module: str, filepath: Optional[str]) -> None:
         self._template = str(self.linter.config.msg_template or self._template)
 
+        # Check to see if all parameters in the template are attributes of the Message
+        template = self._template
+        parameters = re.findall(r"\{(.+?)(:.*)?\}", template)
+        for parameter in parameters:
+            if parameter[0] not in Message._fields:
+                template = re.sub(r"\{" + parameter[0] + r"(:.*?)?\}", "", template)
+        self._template = template
```

Comment:
Just an idea, currently this is down for every module. Can we store the original template and the new one. Then for every subsequent call, compare the "new" `self._template` with the stored `template` and if they match use the fixed one.

Reply:
It might also be a good idea to at least emit a short warning: `Hey, we don't know this option are you sure?`. Otherwise we fail silently.

Reply:
Done and done!